crates/sui-graphql-rpc/src/types/move_registry/on_chain.rs

```diff
@@ -86,23 +82,24 @@ pub(crate) struct AppInfo {
 pub(crate) struct VersionedName {
     /// A version name defaults at None, which means we need the latest version.
     pub(crate) version: Option<u64>,
-    /// The on-chain `Name` object that represents the dot_move name.
+    /// The on-chain `App` object that represents the move registry name.
     pub(crate) name: Name,
 }
 
 /// Attention: The format of this struct should not change unless the on-chain format changes,
 /// as we define it to deserialize on-chain data.
 #[derive(Debug, Serialize, Deserialize, Hash, Clone, Eq, PartialEq)]
 pub(crate) struct Name {
-    pub(crate) labels: Vec<String>,
-    pub(crate) normalized: String,
+    pub(crate) org: Domain,
+    pub(crate) app: Vec<String>,
 }
 
 impl Name {
-    pub(crate) fn new(app_name: &str, org_name: &str) -> Self {
-        let normalized = format!("{}@{}", app_name, org_name);
-        let labels = vec![org_name.to_string(), app_name.to_string()];
-        Self { labels, normalized }
+    pub(crate) fn new(app: &str, org: Domain) -> Self {
```

Comment:
nit: the fact that these parameters are given in the opposite order to the struct is a little confusing.